core/registration.c

```diff
@@ -211,6 +212,10 @@ static void prv_handleRegistrationReply(lwm2m_transaction_t * transacP,
 
             LOG("Registration successful");
         }
+        else if (packet != NULL && packet->code == COAP_231_CONTINUE)
```

Comment:
RFC 7959 allows the final response to come before all blocks have been sent. It is probably better to look at the Block1 option and how much has been sent before processing based on packet->code. It would also be good to check the size in the Block1 option and reduce the block size if requested.

Reply:
Okay, I will look at this.